userspace/falco/application.cpp

```diff
@@ -196,6 +194,9 @@ bool application::run(std::string &errstr, bool &restart)
 		std::bind(&application::daemonize, this),
 		std::bind(&application::init_outputs, this),
 		std::bind(&application::init_clients, this),
+		std::bind(&application::configure_interesting_sets, this),
```

Comment:
I see why you needed to move these actions below in the sequence, but I don't think this would work. Print actions have low responsibility and should not imply unwanted/unexpected behavior of Falco. For example, this means that Falco will attach signal handlers and create the grpc socket (all with chance of failure) before printing the ignored events with `-i`, which we don't want. 

Also, I think that given the change of semantics we don't even need to configure the interesting sets anymore in order to print this information, and the result of these two commands should not be influenced conceptually by the loaded rules (and thus have no real reason to be moved below in the list):
- `-i`, print_ignored_events: From the docs, _"Print all events that are ignored by default (i.e. without the -A flag) and exit"_. In this case, the only events that we ignore without -A are predefined and is the set of IO syscalls, it does not depend on the loaded rules.
- `--list-syscall-events`, print_syscall_events: From the docs, _"List all defined system call events"_. This is a constant set and should not depend on the loaded rules.

Looking at this, I also think the cmdline docs should be updated a little for `-i`.

Reply:
Adjusted (plus updated docs), I liked the suggestion re just printing the static set of I/O syscalls for `-i`, I still check for (now calling them unsupported events) after configuring interesting sets.